packages/ra-core/src/form/useSuggestions.ts

```diff
@@ -84,7 +84,8 @@ const useSuggestions = ({
 
 export default useSuggestions;
 
-const escapeRegExp = value => value.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+const escapeRegExp = value =>
+    value ? value.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') : ''; // $& means the whole matched string
```

Comment:
will fail for value = 0. Would you mind testing if `typeof value === 'undefined'`?

Reply:
well, value should be technically string, so any number would fail?

Reply:
I'm not sure I understand your point. If the user typed the string '0', than the escaped regex will be empty string, which is wrong - it should be '0'. 

Besides, I don't understand in which case the current code fails. Could you give an example of failing case, or, even better, a failing unit test?

Reply:
but, string '0' will pass trough

the value ? value will only stop: undefined, null, false, ''     and replace them with ''